### ISSUE TITLE
LB setters and getters ignore runtime errors

### ISSUE THREAD

**Issue by jngrad** [espressomd/espresso#4354]
**TL;DR:** The exception mechanism in the LB interface is broken. Runtime errors raised in the core are ignored by the Cython code. This is another manifestation of #4219.

**Background information**

When a C++ function throws an exception, the signal can be caught at the Cython level by marking the included C++ function with the modifier `except X`, where `X` is usually `+` (other symbols exists, and they modify how the signal is forwarded from Cython to Python). When an included C++ function is not called directly from Python, but indirectly via an intermediate Cython function, the Cython function needs to be marked `except X` to forward the signal.

These intermediate Cython functions are declared in the `.pxd` file as `cdef inline`, with or without a C++ return type:

* the `cdef inline` statement doesn't mention the return type: then the return type is a Python object, and the signal is wrapped inside that object; when it is sent to the Python interpreter, the signal is deserialized as a Python exception.
* the `cdef inline` statement mentions a C++ return type: then either the signal is encoded as a sentinel value (only for pointers, `int`, `float`, `enum`) if it doesn't collide with a sentinel value from the C++ function (e.g. -1 for `ES_ERROR` and uninitialized global variables...), or it cannot be encoded (any other C++ type, as well as the `void` return type) and the error message is printed to the screen without halting the flow of the program.

For more details, see the Cython documentation section on [error return values](https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#error-return-values) and click on "Cython" to get the version of the documentation that is relevant to us.

**Problem statement**

The LB interface uses intermediate Cython functions which are not marked as `except X`, so they won't forward errors from the core.

Here is a MWE:
```python
import espressomd.lb
system = espressomd.System(box_l=3 * [6.0])
system.time_step = 0.1
lbf = espressomd.lb.LBFluid(visc=1.0, dens=1.0, agrid=1.0, tau=0.1)
system.actors.add(lbf)
node = lbf[0, 0, 0]
system.actors.remove(lbf)
print('start')
print(node.density) # undefined behavior
node.density = 5 # undefined behavior
print('end')
```
Output:
```
$> ./pypresso mwe.py
start
RuntimeError: LB not activated
Exception ignored in: 'espressomd.lb.python_lbnode_get_density'
Traceback (most recent call last):
  File "mwe3.py", line 9, in <module>
    print(node.density)
RuntimeError: LB not activated
0.0
RuntimeError: LB not activated
Exception ignored in: 'espressomd.lb.python_lbnode_set_density'
Traceback (most recent call last):
  File "mwe3.py", line 10, in <module>
    node.density = 5
RuntimeError: LB not activated
end
$> echo $?
0
```
Expected behavior: the script should halt when printing the value of `node.density` and return an error code.

The intermediate Cython function also have a C++ return type, which complicates the way we need to handle the signal. In particular, we cannot use `except +`. Instead, we need to use specializations depending on the return type. For example, functions with `void` return type require `except *`, which introduces overhead in the form of a function call to `PyErr_Occurred()`. When the chosen `except X` specialization isn't correct for the Cython function return type, a malformed Python exception is raised that halts the flow of the program, but crashes the `unittest` framework with a cryptic error message:

```python
...
        with self.assertRaises(RuntimeError):
            print(lbf.viscosity) # raises a malformed RuntimeError object
...
```
```
Traceback (most recent call last):
  File "../testsuite/python/lb.py", line 135, in test_raise_if_not_active
    print(lbf.viscosity)
  File "/usr/lib/python3.8/unittest/case.py", line 227, in __exit__
    self._raiseFailure("{} not raised".format(exc_name))
AttributeError: '_AssertRaisesContext' object has no attribute '_raiseFailure'
```

This is also an issue in the `walberla` branch and all derived branches (`walberla_ubb`, `walberla_ekin`, `lees_edwards`). Some exceptions halt the flow of the program but are still broken, e.g. the node density setter prints a cryptic error message when called in the global namespace of a Python session, but not in a function namespace, nor in the global namespace of a script:

```
>>> node.density = 5
Traceback (most recent call last):
  File "lb.pxd", line 178, in espressomd.lb.python_lbnode_set_density
RuntimeError: LB not activated

The above exception was the direct cause of the following exception:

SystemError: PyEval_EvalFrameEx returned a result with an error set
```

**Solution**

Remove return types from the intermediate Cython functions, such that they can be marked as `except +`. When the return type is a `VectorXd`, convert it to a list. Write python tests to check the exceptions.
